src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardRepository.java

```diff
@@ -383,22 +467,44 @@ protected long findLatestFileNameGeneration(ImmutableMap<String, BlobMetaData> b
          * Loads all available snapshots in the repository
          *
          * @param blobs list of blobs in repository
-         * @return BlobStoreIndexShardSnapshots
+         * @return tuple of BlobStoreIndexShardSnapshots and the last snapshot index generation
          */
-        protected BlobStoreIndexShardSnapshots buildBlobStoreIndexShardSnapshots(ImmutableMap<String, BlobMetaData> blobs) {
-            List<BlobStoreIndexShardSnapshot> snapshots = Lists.newArrayList();
+        protected Tuple<BlobStoreIndexShardSnapshots, Integer> buildBlobStoreIndexShardSnapshots(ImmutableMap<String, BlobMetaData> blobs) {
+            int latest = -1;
+            for (String name : blobs.keySet()) {
+                if (name.startsWith(SNAPSHOT_INDEX_PREFIX)) {
+                    try {
+                        int gen = Integer.parseInt(name.substring(SNAPSHOT_INDEX_PREFIX.length()));
+                        if (gen > latest) {
+                            latest = gen;
+                        }
+                    } catch (NumberFormatException ex) {
+                        logger.warn("failed to parse index file name [{}]", name);
+                    }
+                }
+            }
+            if (latest >= 0) {
+                try (InputStream stream = blobContainer.openInput(SNAPSHOT_INDEX_PREFIX + latest)) {
+                    return new Tuple<>(readSnapshots(stream), latest);
+                } catch (IOException e) {
+                    logger.warn("failed to read index file  [{}]", e, SNAPSHOT_INDEX_PREFIX + latest);
+                }
+            }
+
+
+            List<SnapshotFiles> snapshots = Lists.newArrayList();
```

Comment:
This is fallback right? For older repos that don't have the new snapshot index file?